catalog/dags/common/storage/media.py

```diff
@@ -274,34 +276,34 @@ def _enrich_meta_data(meta_data, license_url, raw_license_url) -> dict:
             )
         return enriched_meta_data
 
-    def _enrich_tags(self, raw_tags) -> list | None:
+    def _enrich_tags(self, raw_tags) -> list[dict] | None:
         """
-        Add provider information to tags.
+        Add provider information to tags, sorted for further consistency between runs,
+        saving on inserts into the DB later.
 
         Args:
-            raw_tags: List of strings or dictionaries
+            raw_tags: List or set of strings.
 
         Returns:
             A list of 'enriched' tags:
             {"name": "tag_name", "provider": self._PROVIDER}
         """
-        if not isinstance(raw_tags, list):
-            logger.debug("`tags` is not a list.")
+        if not isinstance(raw_tags, list) and not isinstance(raw_tags, set):
+            logger.debug("`raw_tags` is not of an accepted type.")
             return None
-        else:
-            return [
-                self._format_raw_tag(tag)
-                for tag in raw_tags
-                if not self._tag_blacklisted(tag)
-            ]
-
-    def _format_raw_tag(self, tag):
-        if isinstance(tag, dict) and tag.get("name") and tag.get("provider"):
-            logger.debug(f"Tag already enriched: {tag}")
-            return tag
-        else:
-            logger.debug(f"Enriching tag: {tag}")
-            return {"name": tag, "provider": self.provider}
+        elif isinstance(raw_tags, list):
+            raw_tags = list(set(raw_tags))
+
+        raw_tags = sorted(raw_tags)
```

Comment:
I think we can do this operation unconditionally, since calling `set` on a set just returns the set (and also `sorted` accepts a set and will always return a list):

```python
In [2]: set({1, 2, 3})
Out[2]: {1, 2, 3}
```

```suggestion
        raw_tags = sorted(set(raw_tags))
```

Reply:
Good, I like it! 💯